keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DockerArtifactSupplier.kt

```diff
@@ -27,11 +27,11 @@ class DockerArtifactSupplier(
   override val eventPublisher: EventPublisher,
   private val cloudDriverService: CloudDriverService,
   override val artifactMetadataService: ArtifactMetadataService
-) : BaseArtifactSupplier<DockerArtifact, DockerVersioningStrategy>(artifactMetadataService) {
+) : BaseArtifactSupplier<DockerArtifact, DockerVersionSortingStrategy>(artifactMetadataService) {
   override val supportedArtifact = SupportedArtifact("docker", DockerArtifact::class.java)
 
-  override val supportedVersioningStrategy =
-    SupportedVersioningStrategy("docker", DockerVersioningStrategy::class.java)
+  override val supportedSortingStrategy =
+    SupportedSortingStrategy("docker", DockerVersionSortingStrategy::class.java)
 
   override fun getArtifactByVersion(artifact: DeliveryArtifact, version: String): PublishedArtifact? {
     return runWithIoContext {
```

Comment:
@gal-yardeni @emjburns Can the call to clouddriver here return images for multiple tags/versions of the same artifact? Something like a `tag = "*"`? If it does, then perhaps we could make a single call instead of N calls before sorting versions.

Reply:
That worked! Please take a look: https://github.com/spinnaker/keel/pull/1652/commits/7acc44793f5a75866f14bbb5182af2ff435015f3